src/routes/helpers/transformers.js

```diff
@@ -0,0 +1,76 @@
+import SchemaHelper from '../../helpers/SchemaHelper'
+import { warning } from '../../utils/logger'
+
+// Used scopes dict
+const scopedContexts = {
+  dc: 'http://purl.org/dc/elements/1.1/',
+  prov: 'https://www.w3.org/TR/prov-o/#',
+  skos: 'http://www.w3.org/2004/02/skos/core#',
+  rdf: 'https://www.w3.org/2000/01/rdf-schema'
+}
+
+/**
+ * Transform document to a JSON-LD structured document
+ * @param {string} type
+ * @param {Object} data
+ * @returns {Object} JSON-LD structured document
+ */
+export const transformJsonLD = (type, data) => {
+  const schemaHelper = new SchemaHelper()
+  const schemaTypes = schemaHelper.getTypeDescription(type).split(',')
+  const prefixes = Object.keys(scopedContexts)
+
+  // Base JSON-LD document
+  const jsonLdData = {
+    '@context': [
+      'https://schema.org/',
+      scopedContexts
+    ],
+    '@type': schemaTypes
+  }
+
+  // Iterate all keys in the data document
+  Object.keys(data).forEach(key => {
+    // Get the property for the current key
+    const property = schemaHelper.findPropertyType(type, key)
+
+    // Use the current value by default
+    let elementValue = data[key]
+
+    // Transform the value when it's a relational property
+    if (elementValue && schemaHelper.isRelationalProperty(property)) {
+      if (Array.isArray(elementValue)) {
+        elementValue = elementValue.map(id => ({
+          '@id': id
+        }))
+      } else if (typeof elementValue === 'string') {
+        elementValue = {
+          '@id': elementValue
+        }
+      }
+    }
+
+    // Iterate over all property scopes
+    property.description.split(',').forEach(uri => {
```

Comment:
I don't like the idea of repurposing the documentation/description fields of the schema to provide semantic meaning. It means that we lose the ability to add documentation here to explain what a field is for and what data to put into it (I'd prefer documentation in the schema rather than a separate document). I also worry that parsing this field could cause problems if bad data is added.
While it would be a bit more work, what about a separate file that provides this mapping? It could be as simple as a json document like `{"Model": {"field": "schema"}}` that we read and parse on startup. A downside of this is that we make a separation of the schema and mapping, which means that we have to remember to update data in two places when we make a change. Thoughts?

Reply:
opened as https://github.com/trompamusic/ce-api/issues/30 for further discussion